### PR TITLE
Configure networking

### PR DESCRIPTION
**Pull request by shilpiBose29**
Updates to network configuration
1) Move -t parameter to dosarray_config
2) Update routes even if there are no updates in iptable
3) Update documentation to emphasize setting up routes in target host

### CODE REVIEW

File: src/dosarray_configure_networking.sh [hunk 1]
```diff
@@ -79,4 +79,4 @@ echo "Running CMD=${CMD}"
 echo
 echo "Please enter sudo password for ${TARGET_PHYSICAL_HOST}"
 
-dosarray_execute_on "${TARGET_PHYSICAL_HOST}" "-t ${CMD}"
```
Comment (niksu) [inline src/dosarray_configure_networking.sh]: This change (removing the "-t") doesn't seem to be related specifically to this commit, judging by the commit message.

Comment (shilpiBose29) [reply src/dosarray_configure_networking.sh]: -t parameter in the network config script interferes with the generation of commands in case of nerds. So I moved it within disarray_execute_on() in config.sh

File: doc/USAGE.md [hunk 2]
```diff
@@ -62,13 +64,18 @@ unnecessary routes have already been deleted.
 ./src/dosarray_configure_network [-r] <physical-host-name>
 ```
 
+NOTE: Configure network on the target machine with the -r option to route packets correctly 
+Applied changes to iptables can be viewed by running `sudo iptable -S`. Similarly, the newly added routes can be viewed by running `ip route show`
```
[the remark this answers is not in the document]
Comment (shilpiBose29) [reply doc/USAGE.md]: Done

File: doc/USAGE.md [hunk 3]
```diff
@@ -73,6 +73,17 @@ Take for example the following route modification command - `sudo route add -net
 ```
 192.168.1.0/24 via 192.168.0.1 dev em1
 ```
+Alternatively, if you run 'ip route get 192.168.1.0' on the host, a correct output would look like:
+```
+192.168.1.0 via 192.168.0.1 dev eno1  src 192.168.0.11
+    cache
```
[the remark this answers is not in the document]
Comment (shilpiBose29) [reply doc/USAGE.md]: It means that all packets with a destination address of 192.168.1.0 will be routed via 192.168.0.1 after adding the routes using network config script

[the remark this answers is not in the document]
Comment (shilpiBose29) [reply doc/USAGE.md]: I think it means there is not cache entry for this route. In case a cache entry existed, the output would look something like this:
192.168.100.17 via 192.168.99.254 dev eth0  src 192.168.99.35 
    cache  users 1 used 326 age 12sec mtu 1500 advmss 1460

[the remark this answers is not in the document]
Comment (shilpiBose29) [reply doc/USAGE.md]: Done

File: config/dosarray_config.sh [hunk 4]
```diff
@@ -74,12 +74,16 @@ export DOSARRAY_CONTAINER_HOST_IDXS=`dosarray_physical_hosts_skip`
...
 function dosarray_execute_on () {
   local HOST_NAME="$1"
   local CMD="$2"
-  ssh <USERNAME>@${HOST_NAME}.<FULLY_QUALIFIED_NAME> -p <SSH_PORT> -t ${CMD}
+  local SSH_PARAM="$3"
+  if [ -z ${SSH_PARAM} ]
+  then
+    ssh <USERNAME>@${HOST_NAME}.<FULLY_QUALIFIED_NAME> -p <SSH_PORT> ${CMD}
+  else
+    ssh <USERNAME>@${HOST_NAME}.<FULLY_QUALIFIED_NAME> -p <SSH_PORT> -t ${CMD}
```
Comment (niksu) [inline config/dosarray_config.sh]: For generality this line could be put as
`ssh <USERNAME>@${HOST_NAME}.<FULLY_QUALIFIED_NAME> -p <SSH_PORT> ${SSH_PARAM} ${CMD}`
since down the line `SSH_PARAM` might need to be a different/additional parameter to `-t`.

Additionally, the `if [ -z ${SSH_PARAM} ]` test wouldn't be necessary, since both cases would map to the line above (since `${SSH_PARAM}`="" in the first case).

**Comment by niksu** [comment]
Transferred development to https://github.com/niksu/dosarray/pull/19
